File: apple/Sources/ReactNativeWatchOS/Modifiers/Tint.swift
import SwiftUI

/// SwiftUI `.tint(_:)`. Sets the tint color applied to controls and accessory
/// content within the view. Resolves `color` via `RNWColorParser`; leaves the
/// view unchanged when the string is unrecognized.
enum RNWTintModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("tint") { view, params, _ in
            guard let colorString = params.string("color"),
                  let color = RNWColorParser.parse(colorString) else {
                return view
            }
            return AnyView(view.tint(color))
        }
    }
}

File: src/modifiers/tint.ts
import { createModifier } from './createModifier';

/// SwiftUI `.tint(_:)`. Sets the tint color applied to controls and
/// accessory content within the view. The color string supports named
/// colors, SwiftUI semantic colors (`'primary'`, `'accent'`), and
/// `#RRGGBB` / `#RRGGBBAA`.
export function tint(color: string) {
  return createModifier('tint', { color });
}
